src/aspire/basis/pswf_2d.py

```diff
@@ -153,18 +153,17 @@ def evaluate_t(self, images):
             to be evaluated.
         :return : The evaluation of the coefficient array in the PSWF basis.
         """
-        images = images.T  # RCOPT
 
-        images_shape = images.shape
+        if not isinstance(images, Image):
+            logger.warning(
+                "FPSWFBasis2D.evaluate_t expects Image instance,"
+                " attempting conversion."
+            )
+            images = Image(images)
 
-        images_shape = (images_shape + (1,)) if len(images_shape) == 2 else images_shape
-        flattened_images = images.reshape(
-            (images_shape[0] * images_shape[1], images_shape[2]), order="F"
-        )
+        flattened_images = images[:, self._disk_mask]
```

Comment:
Are these actually "flattened"?

Reply:
Yes :)

When you take a selection of an ndarray using an ndarray bool mask it usually returns a vector in those dims.  This is because it would generally be "ragged"... no guarantee all rows same length anymore.

In the unit test (one image) this shape is (1,47)... The underlying images data is (1,8,8) and _disk_mask is (8,8).

Reply:
Ah. I see. Thanks for the explanation!